sql/core/src/main/scala/org/apache/spark/sql/execution/command/cache.scala

```diff
@@ -51,32 +54,35 @@ case class CacheTableCommand(
 
     if (storageLevelValue.nonEmpty) {
       sparkSession.catalog.cacheTable(
-        tableIdent.quotedString, StorageLevel.fromString(storageLevelValue.get))
+        tableName, StorageLevel.fromString(storageLevelValue.get))
     } else {
-      sparkSession.catalog.cacheTable(tableIdent.quotedString)
+      sparkSession.catalog.cacheTable(tableName)
     }
 
     if (!isLazy) {
       // Performs eager caching
-      sparkSession.table(tableIdent).count()
+      sparkSession.table(tableName).count()
     }
 
     Seq.empty[Row]
   }
 }
 
-
 case class UncacheTableCommand(
-    tableIdent: TableIdentifier,
+    multipartIdentifier: Seq[String],
     ifExists: Boolean) extends RunnableCommand {
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
-    val tableId = tableIdent.quotedString
-    if (!ifExists || sparkSession.catalog.tableExists(tableId)) {
```

Comment:
This reminds me that we should check `CatalogImpl` and see if n-part name is well supported. Seems like `tableExists` is broken...

Reply:
We don't need to fix `CatalogImpl` completely in this PR, but the related ones should be fixed. For `tableExists`:
```
def tableExists(table: String) = tableExists(parseMultiPartName(table))

def tableExists(db:String, table: String) = tableExists(Seq(db, table))

private def tableExists(name: Seq[String]) ...
```

Reply:
`CatalogImpl` is for session catalog, so it cannot support v2 tables, no?

Reply:
That's a good point, then we can't change `uncacheTable` either...

We need to create a custom `uncacheTable` method in this command as well.

Reply:
Ah, I see your point.

`CatalogImpl.cacheTable` already supports v2 tables since it resolves the table name by `sparkSession.table(tableName)`. But, I think `CatalogImpl.cacheTable`, `CatalogImpl.unCacheTable` should only support v1 tables, and `CacheTableCommand` and `UncacheTableCommand` should go thru `sparkSession.sharedState.cacheManager`, right? I will update accordingly.

Reply:
yup